pkg/pool-stable/contracts/StableMath.sol

```diff
@@ -77,24 +79,29 @@ library StableMath {
             return 0;
         }
 
-        uint256 prevInvariant = 0;
-        uint256 invariant = sum;
-        uint256 ampTimesTotal = amplificationParameter * numTokens;
+        uint256 prevInvariant; // Dprev in the Curve version
+        uint256 invariant = sum; // D in the Curve version
+        uint256 ampTimesTotal = amplificationParameter * numTokens; // Ann in the Curve version
 
         for (uint256 i = 0; i < 255; i++) {
-            uint256 P_D = balances[0] * numTokens;
-            for (uint256 j = 1; j < numTokens; j++) {
-                P_D = Math.div(Math.mul(Math.mul(P_D, balances[j]), numTokens), invariant, roundUp);
+            uint256 D_P = invariant;
+
+            for (uint256 j = 0; j < numTokens; j++) {
+                // (D_P * invariant) / (balances[j] * numTokens)
+                D_P = Math.div(Math.mul(D_P, invariant), Math.mul(balances[j], numTokens), roundUp);
```

Comment:
If I understand this correctly, D_P seems to be the inverse of P_D (or at least close - there seems to be a missing factor of `invariant^2`. If this is true, then likely we'd want to have rounding in the opposite direction as we originally did.

@sergioyuhjtman could you take a look at this?